### ISSUE TITLE
FeatureRequest: Google Drive support

### ISSUE THREAD

**Issue by seaspeak** [pgaskin/NickelMenu#143]
Recently Kobo new firmware officially supports Google Drive (software version 4.37 or higher). Hope to add the feature as for Dropbox.

https://help.kobo.com/hc/en-us/articles/15335985512983-Add-books-to-your-eReader-using-Google-Drive

**Comment by aceflor** [comment]
Same here

**Comment by Dirk71** [comment]
Yes, I also would appreciate Google Drive support, same as it is now for [Dropbox](https://www.mobileread.com/forums/showthread.php?t=344661).

**Comment by heylookltsme** [comment]
+1 

Thanks for nickelmenu! It rules!

**Comment by xdepoorter** [comment]
Yes, I also would appreciate Google Drive support, same as it is now for Dropbox.

**Comment by avichou** [comment]
i'd love that

**Comment by pgaskin** [comment]
I really gotta get the release pipeline migrated to GH Actions so I can get this stuff implemented and do another release...

**Comment by kingo132** [comment]
How to launch Google Drive using nickle menu, it's already there in the system

**Comment by imax9000** [comment]
There's a symbol `_ZN14MoreController11googleDriveEv` which may have the same functionality as `_ZN14MoreController7dropboxEv`, but for Google Drive.

**Comment by imax9000** [comment]
Can confirm that Google Drive integration works perfectly well on Libra 2, but I had to patch hardcoded device model checks in `libnickel.so` for the menu item to show up.

**Comment by kingo132** [comment]
> There's a symbol `_ZN14MoreController11googleDriveEv` which may have the same functionality as `_ZN14MoreController7dropboxEv`, but for Google Drive.

I added this symbol to the code, and the Google setup page appeared. However, we need the URL of googledrive_link_account_start to link the Google Drive account. Since Dropbox used the URL like https://authorize.kobo.com/{region}/{language}/LinkDropbox. So I tried a lot of combos like LinkGoogleDrive, LinkGoogle, LinkDrive, etc. None of them works. We need help with what this link URL is.

**Comment by kingo132** [comment]
One example of the Dropbox URL is https://authorize.kobo.com/us/en/LinkDropbox. When you try this on the computer, it will show an error, but the URL remains the same and won't be redirected.
![QQ_1723261513487](https://github.com/user-attachments/assets/ac7a29c8-3900-4c17-832c-72575c904c2b)
However, if you enter a URL like this: https://authorize.kobo.com/us/en/LinkGoogleDrive. It will not just show this error but redirect it to an error URL: https://authorize.kobo.com/Error/. This could indicate the wrong URL component: "LinkGoogleDrive", which means the guess of this URL path is wrong.
According to Kobo's help page, some devices originally supported Google Drive.
![QQ_1723261769531](https://github.com/user-attachments/assets/2decc29a-63fa-46f4-819b-8bf15d7abfbf)
Therefore, if someone who owns these devices can share the URL of googledrive_link_account_start inside the "Kobo eReader.conf" file, that would be great.

**Comment by ryanplusplus** [comment]
> Therefore, if someone who owns these devices can share the URL of googledrive_link_account_start inside the "Kobo eReader.conf" file, that would be great.

```
googledrive_link_account_start=https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive
```

**Comment by kingo132** [comment]
> > Therefore, if someone who owns these devices can share the URL of googledrive_link_account_start inside the "Kobo eReader.conf" file, that would be great.
> 
> ```
> googledrive_link_account_start=https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive
> ```

Thank you so much! I succeeded in adding the Google Drive functionality to my Kobo Clara Colour!
![QQ_1723305276363](https://github.com/user-attachments/assets/be22ec42-2854-44af-8376-2d1f7d2d1379)
Here is my modified version of NickelMenu; you can use it before the author releases the newest version.
[KoboRoot.zip](https://github.com/user-attachments/files/16570138/KoboRoot.zip)
Unzip it and put it into .kobo folder, and as you know, modify the .kobo/Kobo/Kobo eReader.conf
```
googledrive_link_account_start=https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive
kobo_googledrive_link_account_enabled=True
```
Then add this to your NickelMenu
```
menu_item :main    :godrive            :nickel_open        :library   :godrive
```

All the code I modified is in the action_cc.cc file
![QQ_1723305577617](https://github.com/user-attachments/assets/11e12272-5f8c-4013-a8dd-3909e35784b1)

Thank you so much for providing this information!

**Comment by imax9000** [comment]
Also see this https://github.com/pgaskin/kobopatch-patches/pull/139 if you're using kobopatch

**Comment by ryanplusplus** [comment]
> Thank you so much for providing this information!

No problem, I'm glad I could help! Thanks for making that build!

**Comment by heylookltsme** [comment]
This is awesome, thanks for sharing! 

@kingo132 Could you open a pull request for your updates? 🙏🏻

**Comment by Krabbis07** [comment]
> > > Therefore, if someone who owns these devices can share the URL of googledrive_link_account_start inside the "Kobo eReader.conf" file, that would be great.
> > 
> > 
> > ```
> > googledrive_link_account_start=https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive
> > ```
> 
> Thank you so much! I succeeded in adding the Google Drive functionality to my Kobo Clara Colour! ![QQ_1723305276363](https://private-user-images.githubusercontent.com/46492291/356822291-be22ec42-2854-44af-8376-2d1f7d2d1379.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5NDgwNzAsIm5iZiI6MTcyNjk0Nzc3MCwicGF0aCI6Ii80NjQ5MjI5MS8zNTY4MjIyOTEtYmUyMmVjNDItMjg1NC00NGFmLTgzNzYtMmQxZjdkMmQxMzc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDE5NDI1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlMjQ2MTg4MzE3ODM1ZjA0MGVkOWJmZjczYmRjYzE1YzI3ZDhlYzk3MzBlYmI3M2IxZjVkMzczZWJjYTE1Y2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CEdneLwUq72DJ5RtvgMqNCaBvtDpmi_i8F9q9p2FJGQ) Here is my modified version of NickelMenu; you can use it before the author releases the newest version. [KoboRoot.zip](https://github.com/user-attachments/files/16570138/KoboRoot.zip) Unzip it and put it into .kobo folder, and as you know, modify the .kobo/Kobo/Kobo eReader.conf
> 
> ```
> googledrive_link_account_start=https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive
> kobo_googledrive_link_account_enabled=True
> ```
> 
> Then add this to your NickelMenu
> 
> ```
> menu_item :main    :godrive            :nickel_open        :library   :godrive
> ```
> 
> All the code I modified is in the action_cc.cc file ![QQ_1723305577617](https://private-user-images.githubusercontent.com/46492291/356822623-11e12272-5f8c-4013-a8dd-3909e35784b1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5NDgwNzAsIm5iZiI6MTcyNjk0Nzc3MCwicGF0aCI6Ii80NjQ5MjI5MS8zNTY4MjI2MjMtMTFlMTIyNzItNWY4Yy00MDEzLWE4ZGQtMzkwOWUzNTc4NGIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDE5NDI1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhYjc3N2NhOThjODY5YjgzMWU5ODcwY2Y3ZGI3NDY4N2ZmMDZhNjJmMGQzN2QyMThiNTdlOTJmOWU4ZjEwNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hooJ7fsqlzAb7aRmN93f-bY1u-iWuQck43FmTVHYsQo)
> 
> Thank you so much for providing this information!

Hi, new to github and dealing with code, so was wondering if you could provide some clarification about how you enabled google drive? 

I managed successfully with Dropbox and did the same steps with the information you provided for Google Drive i.e. edited the Kobo eReader.conf and added menu item to the nickelmenu. However, when I try to open it on my Kobo I get the following popup "unknown view 'godrive' (in 'library:godrive') (check failed: sym_f)

The only missing piece I can think of is what was mentioned earlier as "a symbol" _ZN14MoreController11googleDriveEv
which I think is related to your modified code in the action_cc.cc file screenshot. Is this something I need to edit too/something that was included in your modified KoboRoot file and not in the original? 

Will I need to uninstall my nickelmenu and install your modified version instead, or can I edit it on my end? (I however can't find a file in my kobo's folders called action_cc.cc to edit!)

Any guidance is appreciated! 🙏🏼

**Comment by brianpipa** [comment]
@Krabbis07 sounds like you didn't install the modified Nickelmenu - look at the last comment by kingo132 - he has a link to a KoboRoot.zip - install that Nickelmenu update then it will work. I did it the other day on my Kobo mini running Kobo Glo firmware (4.38) and it worked beautifully

**Comment by Dirk71** [comment]
The above mentioned KoboRoot.zip for the updated NickleMenu doesn't work. I have installed it and still the "godrive" isn't known by NickleMenu - I get a failure message....._**unknown view 'godrive' (in libary:godrive')**_

**Comment by niceguysean** [comment]
I have tried and failed miserably to install this on my partners Kobo Clara Color. Followed the instructions above, Unzip and installed KoboRoot.zip and put it into .kobo folder, and modified the .kobo/Kobo/Kobo eReader.conf, but Nickelmenu does not show up when I reboot the device.
Kobo Clara Color 4.40.23081
Any help gratefully recieved.

**Comment by Fatima-Fatehullah** [comment]
Im so lost I tried everything but its showing up as godrive is unavailable. Also I'm new to this so can anyone tell me how to update nicklemenu. I have a clara 2e and I just want to add google drive to it. any help is appreciated.

**Comment by arcanemuse** [comment]
Thank you for that modified nickel menu.  I have Google Drive now on my old Aura H2O.  Works great!

**Comment by JMLucas96** [comment]
Hi, any updates on this? How can I put NM in my kobo libra 2 and google drive patch? I'm new on that thread...
Thank you very much!! 🙏

**Comment by TomasHurtz** [comment]
Hi there please help !

I have kobo clara color and trying to get Google Drive functional:

1) Downloaded the [KoboRoot.zip](https://github.com/user-attachments/files/16570138/KoboRoot.zip)
2) Unzipped it and added the .tgz to the kobo folder, 
3) Rebooted and can see the Nickelmenu 
4) Plugged back into PC and see the  .adds folder - but I don't see any file called "action_cc.cc" 
I only have a file called doc - so where to add the lineL:

menu_item :main    :godrive            :nickel_open        :library   :godrive

**Comment by davidoclubb** [comment]
Can confirm this works perfectly for the Kobo Libra H20. Thankyou everybody who contributed 🙏

**Comment by Smog418** [comment]
> Hi there - here is some help for those who need it (added Dec 27th 2024). If you are a beginner, this might seem a bit daunting but actually it is very easy - stick with it and you'll have Google Drive enabled in no time!
> 
> I have Kobo Clara Color and wanted to get Google Drive functional:
> 
> On your PC (or mac) follow the steps below. Note: I recommend you use Notepad++ a free utility to edit the files
> 
> 1. Download the [KoboRoot.zip](https://github.com/user-attachments/files/16570138/KoboRoot.zip) to your computer (Note: this file is linked from the one above from [kingo132](https://github.com/kingo132) - I am not sure if there is a new Master from [@pgaskin](https://github.com/pgaskin) - if there is, perhaps it could be added below, thanks ).  Now go find the downloaded KoboRoot.zip folder, unzip that downloaded folder and inside you should be able to see a file called KoboRoot.tgz
> 2. Connect your Kobo via USB to the computer and go find it on your computer folder list of drives.  Now move the .tgz folder from step 1, to your E:**.kobo** folder (your drive letter might be different).  Note the ".kobo" folder is in the root directory, next to another folder called .kobo-images.  Now disconnect the Kobo from the computer (Windows eject function is recommended) and then unplug the USB cable from the Kobo device - the Kobo device should automatically reboot itself and load the new tgz config.  Wait ~1 minute and once it has rebooted, you should see the NickelMenu next to your other menu items at the bottom right. If not, you've done something wrong and will have to try again!
> 3. Now we will configure the Google Drive menu item.  Connect your Kobo device via USB to the computer again and go find it on your computer folder list of drives.  Then open the file (using notepad++) located at: E:.kobo\Kobo\Kobo eReader.conf
> 
> We need to modify some lines as follows:
> 
> * Find:
>   googledrive_link_account_start=
> * and replace with :
>   googledrive_link_account_start=[https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive](https://authorize.kobo.com/%7Bregion%7D/%7Blanguage%7D/linkcloudstorage/provider/google_drive)
> * Scroll down a few lines and find
>   kobo_googledrive_link_account_enabled=False
> * and change to
>   kobo_googledrive_link_account_enabled=True
> 
> Save that file
> 
> 4. Now navigate back up to the Kobo eReader root directory folder again - there should be a new folder called ".adds"  - click into that and then into "nm" and you'll see a file called "doc" - this is a help file, read it if you're curious and want to see all the possible commands to use.
> 5. In this nm folder, we need to make a new TXT file to hold the menu items for our new NickelMenu.  You can call it what you like, for example "command.TXT" - the extension doesn't matter.  I made my new file as a TXT document, using Notepad ++  and in this new file, add these lines:
> 
> ```
> # Add a comment in the file, like this using the hash at the front
> # Menu items for the NickelMenu
> # Created on [add your date]
> # help article here: https://github.com/pgaskin/NickelMenu/issues/143#issuecomment-2563299861
> 
> menu_item :main    :Google Drive  :nickel_open   :library   :godrive
> 
> # A few extra quick access commands - you can take these or leave these as you wish
> menu_item :main     :Sleep              :power :sleep
> menu_item :main     :Shutdown           :power :shutdown
> menu_item :main     :Reboot             :power :reboot
> ```
> 
> These will all appear in the NickelMenu (You can see more example commands in the "doc" help file.)
> 
> 6. Save the new "command.txt" file in the nm folder and then disconnect the Kobo device from your computer and disconnect the USB cable, and you should now see the new menu items under the NickelMenu.  If you like you can reboot the device using your new NickelMenu option :-)
> 7. Open up the Google Drive item and wait a moment for it to connect (you need to be connected to wifi).  You should then be presented with a screen from Google Drive with "get started" - simply follow the instructions to complete the login connection (you will also need to use your computer or phone to complete the pairing using the presented code to confirm authorisation).
> 
> Once you complete this part, Google Drive folder should now appear on your Kobo device as a folder. You can then add eBooks epubs and pdfs - what you like - to you Google Drive folder (e.g. using your phone or computer) and then easily access them, read and save them on your Kobo device!
> 
> Troubleshooting:
> 
> 1. If you don't see the NickelMenu after step 1, you'll need to go through that step again.
> 2. If after adding the new menus and items, and then opening up the Google Drive connection - if you don't see the Google Drive logo and Get Started screen, check you have followed all the steps (especially step3) and that you are connected to a decent wifi network.
> 
> This has been tested on several separate devices - and it works fine ! This feature should really be available on the native device, I guess Kobo wanted us to buy the more expensive Libre!
> 
> Credits: thanks to all of the above authors who provided the good intel, i just composed it into a easy to follow instruction. Please comment if this helped you, or if you ran into troubles?

Is it supposed to work with the old Kobo Touch? I tried and everything worked, until I tried to connect to the google drive, it no longer detects wifi. I see wifi networks, but I get an error message when I try to connect to them.

**Comment by TomasHurtz** [comment]
Hey @Smog418 

> but I get an error message when I try to connect to them.

what error message exactly - supply details.

**Comment by ludimar75** [comment]
Hello, i have a Kobo Clara Colour fw 4.41.23145 , i installed the modified koboroot.tgz for google drive supporting and added and replaced

[stack_00.log](https://github.com/user-attachments/files/18810691/stack_00.log)

 in koboconfig:

googledrive_link_account_start=https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive

kobo_googledrive_link_account_enabled=True

and in nickel menu config : menu_item :main    :Google Drive  :nickel_open   :library   :godrive

but despite this when i authorize kobo to google drive writing the authorizing code then the device reboot itself and goes in loop in the prodecure, i attach the log error file, so should be useful to understand what is happened

**Comment by mixels** [comment]
@TomasHurtz Thank you for this guide and NickelMenu update. However when I follow the guide, I get an infinite loading screen when clicking the Google Drive button in the new menu on the Clara BW v. 4.41.23145. Do you know what might cause this or have any suggestions please?

**Comment by Varencienne** [comment]
@TomasHurtz 
I am very new to this and followed your instructions but when I click "Google Drive" in the NickelMenu, it gives me an error of "unknown view 'godrive' (in 'library:godrive') (check failed: sym_f)".  other people seem to have this issue and not sure how to fix it. I have tried doing a factory reset on the Clara BW and starting your instructions again. Used Notepad++. Not sure what I am doing wrong. Thanks for any help!

**Comment by mixels** [comment]
> [@TomasHurtz](https://github.com/TomasHurtz) I am very new to this and followed your instructions but when I click "Google Drive" in the NickelMenu, it gives me an error of "unknown view 'godrive' (in 'library:godrive') (check failed: sym_f)". other people seem to have this issue and not sure how to fix it. I have tried doing a factory reset on the Clara BW and starting your instructions again. Used Notepad++. Not sure what I am doing wrong. Thanks for any help!

Try `gdrive` instead of `godrive`.

**Comment by Varencienne** [comment]
Thanks @mixels for the suggestion. I tried that and it still doesn't work. I am on software version 4.41.23145 if that helps at all. I also tried to uninstall the NickelMenu per the instructions of creating a new doc in "nm" to uninstall it and it didn't work. Factory reset of the Kobo also did not remove the NickelMenu and now it just says configerror. Clearly I did something very wrong and I don't know how to fix it now ><  Any suggestions at all would be helpful but understand it is hard to do without seeing the Kobo itself.

**Comment by TomasHurtz** [comment]
Hello 

I hope you can solve your problem. I am no expert. It's been a while since I did this, but I did it with several different devices and made the write up as I went.  It all worked fine and was correct at that time, so I am not sure if something has changed - but would recommend that you make sure to go through all of the steps very carefully to ensure no mistake is made.  It's also important to be connected to good wifi when making the google dive connection (don't use hotspots, vpn etc).  To provide more help, I am adding screenshots, but I can't say if the file path etc will be exactly the same for you, but should give some visual clues perhaps.

![Image](https://github.com/user-attachments/assets/19afa5fd-c684-4ca1-8b0a-60a119ece13c)

![Image](https://github.com/user-attachments/assets/2e5dc7ba-614d-42c2-9460-11c8c0693624)
![Image](https://github.com/user-attachments/assets/f55238fb-11c8-4506-8f79-9b5ca0c2c9dc)
![Image](https://github.com/user-attachments/assets/7580ac56-aaa7-4f39-8256-d2b578728a42)
![Image](https://github.com/user-attachments/assets/f25d47cf-38da-43df-84b7-a74fd491c2b1)
![Image](https://github.com/user-attachments/assets/4d8d6197-684e-4f1d-9a22-3cc67b48cf67)
![Image](https://github.com/user-attachments/assets/07941ff9-e437-4285-a69d-3eb7d71d3e5f)

**Comment by mixels** [comment]
Yes thanks Tomas. Unfortunately the PR that that build was taken from is
still not merged with NickleMenu.

However I do want to point out that the version awaiting merge to
NickleMenu uses `gdrive` as the location name and not `godrive`, so the
command that works with the above posted build will NOT work, presumably,
with the future release of NickleMenu that contains this feature (assuming
it gets merged).

Because Tomas's build was uploaded as an attachment, it is a fixed resource
that does not contain the latest changes under the PR. As a result the name
to use for that specific build is indeed `godrive`.

**Comment by TomasHurtz** [comment]
hello @mixels thanks for your contribution.

I am not sure if you are someone who is working on the new PR ?

> Yes thanks Tomas. Unfortunately the PR that that build was taken from is
> still not merged with NickleMenu.

Indeed. As far as I am aware, my previous write up is still the correct - current one.

However to the developers - I would recommend that you retain the use of the "godrive" in order to preserve backwards compatibility. No sense in changing it for the sake of a single letter and this change then break all of the older devices who may not update.

**Comment by Varencienne** [comment]
Thank you @TomasHurtz! I know this may seem silly and probably intuitive for all of you - but when I copied the words EXACTLY as you had them in your screenshot for the command file, it worked. I had used the text above from your post in Dec 2024 and didn't change anything there and it never worked before. I am not sure if all those extra words and # matter, but it works now :)

Thanks for your patience with my (silly) questions!

**Comment by TomasHurtz** [comment]
> Thank you [@TomasHurtz](https://github.com/TomasHurtz)! I know this may seem silly and probably intuitive for all of you - but when I copied the words EXACTLY as you had them in your screenshot for the command file, it worked. I had used the text above from your post in Dec 2024 and didn't change anything there and it never worked before. I am not sure if all those extra words and # matter, but it works now :)


Yes it is very possible that the text, when copy and pasted, contains rogue hidden characters, especially if you use Microsoft word or other similar document program because they add code you can't see and that causes trouble for code reading programs. 

That is why it is a good idea to use notepad++ because it strips out the hidden characters when you copy from GitHub and then paste into notepad++ and then save it on your kobo, it should then be clean of the Unicode or any other hidden characters that might cause troubles.

Glad it's working for you. 😁😁 Please give my solution a like so others can know it works.

**Comment by avichou** [comment]
is there any idea as to when the next update will release ?

**Comment by mcabe** [comment]
May 2025 and this doesn't work for me. When clicking the item on the kobo I get the error "unknown view 'godrive' (in 'library:godrive') (check failed:sym_f)".

No amount of changing the line makes it work:

`menu_item :main    :Google Drive  :nickel_open   :library   :godrive`

**Comment by SpiderJonny** [comment]
> May 2025 and this doesn't work for me. When clicking the item on the kobo I get the error "unknown view 'godrive' (in 'library:godrive') (check failed:sym_f)".
> 
> No amount of changing the line makes it work:
> 
> `menu_item :main :Google Drive :nickel_open :library :godrive`

Same here.  

I've also noticed that the value of googledrive_link_account_start= that the instructions say to add in the Kobo eReader.conf file — ie,

[https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive](https://authorize.kobo.com/%7Bregion%7D/%7Blanguage%7D/linkcloudstorage/provider/google_drive)

 — gives the following error:

**Server Error**
 
404 - File or directory not found.
The resource you are looking for might have been removed, had its name changed, or is temporarily unavailable.

ie, it's not just a Rakuten error, like https://github.com/pgaskin/NickelMenu/issues/143#issuecomment-2278965921   

So, I'm wondering if it's now the correct link for googledrive_link_account_start=  ?

**Comment by mcabe** [comment]
> I've also noticed that the value of googledrive_link_account_start= that the instructions say to add in the Kobo eReader.conf file — ie,
> 
> [https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive](https://authorize.kobo.com/%7Bregion%7D/%7Blanguage%7D/linkcloudstorage/provider/google_drive)
> 
> — gives the following error:
> 
> **Server Error**
> 
> 404 - File or directory not found. The resource you are looking for might have been removed, had its name changed, or is temporarily unavailable.

I noticed the 404 as well. Wondering if it’s a new link or if they have changed the setup process to not allow this method. I was honestly surprised it requires authenticating through their site.

**Comment by exactmultiple3425** [comment]
> > May 2025 and this doesn't work for me. When clicking the item on the kobo I get the error "unknown view 'godrive' (in 'library:godrive') (check failed:sym_f)".
> > No amount of changing the line makes it work:
> > `menu_item :main :Google Drive :nickel_open :library :godrive`
> 
> Same here.
> 
> I've also noticed that the value of googledrive_link_account_start= that the instructions say to add in the Kobo eReader.conf file — ie,
> 
> [https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive](https://authorize.kobo.com/%7Bregion%7D/%7Blanguage%7D/linkcloudstorage/provider/google_drive)
> 
> — gives the following error:
> 
> **Server Error**
> 
> 404 - File or directory not found. The resource you are looking for might have been removed, had its name changed, or is temporarily unavailable.
> 
> ie, it's not just a Rakuten error, like [#143 (comment)](https://github.com/pgaskin/NickelMenu/issues/143#issuecomment-2278965921)
> 
> So, I'm wondering if it's now the correct link for googledrive_link_account_start= ?

You need to use the modified version of NickelMenu in https://github.com/pgaskin/NickelMenu/issues/143#issuecomment-2282194408

**Comment by mcabe** [comment]
> You need to use the modified version of NickelMenu in [#143 (comment)](https://github.com/pgaskin/NickelMenu/issues/143#issuecomment-2282194408)

I can't get that version working at all. Unzipping that KoboRoot.zip and putting into the .kobo doesn't do anything. Do I need to make sure its a .tgz? Uninstalling the official NickelMenu to attempt to reinstall is also a pain since the "uninstall" file in nm doesn't remove everything.


> Note that you cannot visit that URL in a web browser - it must be added to the config file. If you visit the link in a web browser you will get that 404
> 
> 1. Are you getting the error from within the kobo device?
> 2. What device do you have?
> 3. Which instructions have you followed?
> 4. Have you followed the instructions EXACTLY and double checked?
> 
> It still works fine for me, so curious to know why it's not working for you.


I've followed every instruction in this thread and resinstalled multiple times. NickelMenu can install and I can add most things to the config menu, and I've edited the two lines in Koboereader.conf to add Google Drive(=True and URL). Google Drive does not work and gives the error "unknown view 'godrive' (in 'library:godrive') (check failed:sym_f)".

SpiderJonny is having the same problem so this may be an issue with new kobo's trying to authorize Google Drive.

**Comment by SpiderJonny** [comment]
> > You need to use the modified version of NickelMenu in [#143 (comment)](https://github.com/pgaskin/NickelMenu/issues/143#issuecomment-2282194408)
> 
> I can't get that version working at all. Unzipping that KoboRoot.zip and putting into the .kobo doesn't do anything. 

Same here,

> Uninstalling the official NickelMenu to attempt to reinstall is also a pain since the "uninstall" file in nm doesn't remove everything.

What doesn't it remove?  I have just uninstalled NM and the added NM menu has (obviously) disappeared, but I haven't checked for "leftovers".

> I've followed every instruction in this thread and resinstalled multiple times. NickelMenu can install and I can add most things to the config menu, and I've edited the two lines in Koboereader.conf to add Google Drive(=True and URL). Google Drive does not work and gives the error "unknown view 'godrive' (in 'library:godrive') (check failed:sym_f)".

Ditto.

Thanks

SJ

**Comment by exactmultiple3425** [comment]
> > You need to use the modified version of NickelMenu in [#143 (comment)](https://github.com/pgaskin/NickelMenu/issues/143#issuecomment-2282194408)
> 
> I can't get that version working at all. Unzipping that KoboRoot.zip and putting into the .kobo doesn't do anything. Do I need to make sure its a .tgz? Uninstalling the official NickelMenu to attempt to reinstall is also a pain since the "uninstall" file in nm doesn't remove everything.
> 
> > Note that you cannot visit that URL in a web browser - it must be added to the config file. If you visit the link in a web browser you will get that 404
> > 
> > 1. Are you getting the error from within the kobo device?
> > 2. What device do you have?
> > 3. Which instructions have you followed?
> > 4. Have you followed the instructions EXACTLY and double checked?
> > 
> > It still works fine for me, so curious to know why it's not working for you.
> 
> I've followed every instruction in this thread and resinstalled multiple times. NickelMenu can install and I can add most things to the config menu, and I've edited the two lines in Koboereader.conf to add Google Drive(=True and URL). Google Drive does not work and gives the error "unknown view 'godrive' (in 'library:godrive') (check failed:sym_f)".
> 
> SpiderJonny is having the same problem so this may be an issue with new kobo's trying to authorize Google Drive.

There is a tgz file directly inside the zip, you have to put the tgz into .kobo. Dont use built in decompressor if you are using Mac

**Comment by exactmultiple3425** [comment]
> > There is a tgz file directly inside the zip, you have to put the tgz into .kobo. Dont use built in decompressor if you are using Mac
> 
> Thanks, I put it into terminal and got the .tgz. Once I put this modified koboroot.tgz into the .kobo I no longer get the error above.
> 
> When I click on the NickelMenu tab with Google Drive now I no longer get the same error. It opens a launcher for Google Drive, but clicking "Get Started" doesn't open anything. It either spins forever or opens a white page.
> 
> ![Image](https://github.com/user-attachments/assets/df87ac60-b0cc-49fd-b5c3-476adc2610f2) ![Image](https://github.com/user-attachments/assets/2b4c3d2a-0800-41b9-a406-eb7d32e046ce)

Make sure to follow the instructions regarding updating the Kobo eReader.conf

Remember, UPDATE the config in the existing file, NOT ADDING the new key value pair

https://github.com/pgaskin/NickelMenu/issues/143#issuecomment-2282194408

**Comment by mcabe** [comment]
> Remember, UPDATE the config in the existing file, NOT ADDING the new key value pairs
> 
> [#143 (comment)](https://github.com/pgaskin/NickelMenu/issues/143#issuecomment-2282194408)

Yes, I've modified "Kobo eReader.conf" to have the two Google lines:

`googledrive_link_account_start="googledrive_link_account_start=https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive"`

and 

'kobo_googledrive_link_account_enabled=True'

**Comment by exactmultiple3425** [comment]
> > Remember, UPDATE the config in the existing file, NOT ADDING the new key value pairs
> > [#143 (comment)](https://github.com/pgaskin/NickelMenu/issues/143#issuecomment-2282194408)
> 
> Yes, I've modified "Kobo eReader.conf" to _change_ the two lines referencing Google:
> 
> `googledrive_link_account_start="googledrive_link_account_start=https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive"`
> 
> and
> 
> 'kobo_googledrive_link_account_enabled=True'
> 
> The kobo can not pass the 'Get Started' page above.

You are an idiot, it should be

googledrive_link_account_start=https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive

**Comment by TomasHurtz** [comment]
> You are an idiot....

Hey please refrain from insults. We are all learning here and it's easy to miss stuff especially if a newbie. Thanks

>googledrive_link_account_start=[https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive](https://authorize.kobo.com/%7Bregion%7D/%7Blanguage%7D/linkcloudstorage/provider/google_drive)

Yes try that and see if it works now.

Also a note, I found during my experiments that it matters what wifi you are on,.... Make sure it's stable. Ideally the same wifi you'll be using with the device.  Maybe be irrelevant but I had trouble on one wifi connection when I switched to another it worked flawlessly.  Remember you're a hacker now... It's trial and error, this is art not a science as such....  😆

**Comment by exactmultiple3425** [comment]
> > You are an idiot....
> 
> Hey please refrain from insults. We are all learning here and it's easy to miss stuff especially if a newbie. Thanks
> 
> > googledrive_link_account_start=[https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive](https://authorize.kobo.com/%7Bregion%7D/%7Blanguage%7D/linkcloudstorage/provider/google_drive)
> 
> Yes try that and see if it works now.
> 
> Also a note, I found during my experiments that it matters what wifi you are on,.... Make sure it's stable. Ideally the same wifi you'll be using with the device. Maybe be irrelevant but I had trouble on one wifi connection when I switched to another it worked flawlessly. Remember you're a hacker now... It's trial and error, this is art not a science as such.... 😆

I can’t give sympathy to people who was asked to RTFM twice and still failed to do so

**Comment by Dirk71** [comment]
@tomkimberlin Thank you for this comprehensive description. My Cobo Clara BW (latest firmware: 4.42.23296) got stuck in a bootloop after flashing Nickelmenu. Especially the Kobo Clara BW/Color are prone to issues using patches. My Libra 2 is more robust in this regard. The latest firmwares 4.4x seem to have essential changes that require new scripts for patching.

**Comment by avichou** [comment]
> Hello good afternoon! 
> 
> I have followed all the steps to install google drive on my Kobo Clara, and I do see the 'Google Drive', 'Sleep', 'Shutdown' and 'Reboot' options in the NickelMenu. However, when I click on Google Drive, I get the following error: Google Drive unknown view 'godrive' (in 'library: godrive') (check failed: sym_f). Does anyone know what I can do to fix this? 
> 
> Thanks a lot in advance!

Did you manage to fix it ?

**Comment by avichou** [comment]
> > > Therefore, if someone who owns these devices can share the URL of googledrive_link_account_start inside the "Kobo eReader.conf" file, that would be great.
> > 
> > 
> > ```
> > googledrive_link_account_start=https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive
> > ```
> 
> Thank you so much! I succeeded in adding the Google Drive functionality to my Kobo Clara Colour! ![QQ_1723305276363](https://github.com/user-attachments/assets/be22ec42-2854-44af-8376-2d1f7d2d1379) Here is my modified version of NickelMenu; you can use it before the author releases the newest version. [KoboRoot.zip](https://github.com/user-attachments/files/16570138/KoboRoot.zip) Unzip it and put it into .kobo folder, and as you know, modify the .kobo/Kobo/Kobo eReader.conf
> 
> ```
> googledrive_link_account_start=https://authorize.kobo.com/{region}/{language}/linkcloudstorage/provider/google_drive
> kobo_googledrive_link_account_enabled=True
> ```
> 
> Then add this to your NickelMenu
> 
> ```
> menu_item :main    :godrive            :nickel_open        :library   :godrive
> ```
> 
> All the code I modified is in the action_cc.cc file ![QQ_1723305577617](https://github.com/user-attachments/assets/11e12272-5f8c-4013-a8dd-3909e35784b1)
> 
> Thank you so much for providing this information!

hey, thank you for this. where should I input the line in eReader.conf exactly? like , under what section? General], or somewhere else ?

I also see that google drive link account is "false" for some reason, should i go on an modify it ? 

<img width="378" height="192" alt="Image" src="https://github.com/user-attachments/assets/2e6885a5-1300-47e6-85a6-32d2ac152dbd" />

**Comment by redphx** [comment]
> ![image](https://github.com/user-attachments/assets/91563153-c8ea-4896-8189-8e628a1f80c8)
> 
> Throwing an error for me (after I updated to nickel menu 6.0

it's `gdrive`, not `godrive`

**Comment by Pallandos** [comment]
I can confirm that it is `gdrive` and **not** `godrive` (tested on my Clara BW on 2025/12/14)

However, i am stuck in an eternal loading loop after entering the menu, even connected to a gb wifi.

**Comment by pgaskin** [comment]
@redphx the fact that NM doesn't include the config change is intentional, and if you're thinking about it, please don't write a mod for it either.

**Comment by redphx** [comment]
@pgaskin noted. You can delete my previous issue if you want.

**Comment by NerdyyGuy** [comment]
> I can confirm that it is `gdrive` and **not** `godrive` (tested on my Clara BW on 2025/12/14)
> 
> However, i am stuck in an eternal loading loop after entering the menu, even connected to a gb wifi.

Hey there, we're you ever able to get passed the loading screen? I added it to my Nickel Menu successfully, clicked on it, got to the google drive screen where it shows a button to get started but after pressing it I also just get stuck on a loading screen...

Using a Kobo Clara BW

**Comment by Leo3418** [comment]
> I can confirm that it is `gdrive` and **not** `godrive` (tested on my Clara BW on 2025/12/14)
> 
> However, i am stuck in an eternal loading loop after entering the menu, even connected to a gb wifi.

For those encountering the same issue, I solved it by rebooting the device.

**Comment by crazybookreads-create** [comment]
Does this work with Kobo Libra Color while in Sideloaded Mode? 

*Edit: It does, I have it working
